tsconfig.json

```diff
@@ -1,9 +1,10 @@
 {
   "compilerOptions": {
     "outDir": "dist",
-    "lib": ["es2016"],
-    "declaration": true,
-    "sourceMap": true
+    "moduleResolution": "node",
+    "target": "es2018",
```

Comment:
I am no expert but shouldn't it be esnext, and then bundler could transpile it if necessary?

I recently saw an article about this:
https://github.com/frehner/modern-guide-to-packaging-js-library#target-modern-browsers

Reply:
Hi, thanks for the article! Interesting, I may be saving it for later.
**Quick disclaimer:** I am no expert either on this topic, I am still trying to figure it out 😅 

Anyway, so, regarding your query. It is true that setting it to `esnext` would allow those using a bundler to take advantage of the latest features and then transpile it to older versions, but this library is not targeted to browsers only, but also to node in general, and to project using simple JS, so without the use of TS.

For the ES version (`es2018`) I selected, I did so following this recommendations: https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping, aiming for node v10 as lowest.

Since in this project the code is tested on node12 as lowest node version, we could bump the config to `es2019` or even `es2020`, since now node 14 is the lowest LTS version of node. But I do not know if it is worth it to go to `esnext`, maybe there will be issues with older codebases...

But this is something I would let decide the maintainer of the library.

Reply:
I also agree that setting to esnext would be ideal, this way we can have the TS compiler do as little transpilation as possible, making it the responsibility of the bundler, which is probably what we want.

Reply:
Ok then, I can change both lib and target it to `esnext` if you prefer it this way

Reply:
sounds good, except I think lib should stay as it is, we were only talking about target

Reply:
Ok, got it!